x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/planner/QueryTranslator.java

```diff
@@ -735,7 +724,14 @@ protected QueryTranslation asQuery(In in, boolean onAggs) {
                 else {
                     Query q = null;
                     if (in.value() instanceof FieldAttribute) {
-                        q = new TermsQuery(in.source(), ne.name(), in.list());
+                        FieldAttribute fa = (FieldAttribute) in.value();
+                        String name = fa.name();
+                        // equality should always be against an exact match
+                        // (which is important for strings)
+                        if (fa.isInexact()) {
+                            name = fa.exactAttribute().name();
+                        }
+                        q = new TermsQuery(in.source(), name, in.list());
```

Comment:
My personal preference is for `q = new TermsQuery(in.source(), fa.isInexact() ? fa.exactAttribute().name() : fa.name(), in.list());`.